pinecone/core/grpc/index_grpc.py

```diff
@@ -4,20 +4,30 @@
 import logging
 from abc import ABC, abstractmethod
 from functools import wraps
-from typing import NamedTuple, Optional, Dict
+from typing import NamedTuple, Optional, Dict, Iterable
 
 import grpc
 import certifi
+from google.protobuf import json_format
 
 from pinecone.config import Config
 from pinecone.core.grpc.protos.vector_column_service_pb2_grpc import VectorColumnServiceStub
 from pinecone.core.grpc.protos import vector_service_pb2, vector_column_service_pb2
-from pinecone.core.utils import _generate_request_id
+from pinecone.core.utils import _generate_request_id, dict_to_proto_struct, fix_tuple_length, proto_struct_to_dict
 from pinecone.core.utils.sentry import sentry_decorator as sentry
 from pinecone.core.grpc.protos.vector_service_pb2_grpc import VectorServiceStub
 from pinecone.core.grpc.retry import RetryOnRpcErrorClientInterceptor, RetryConfig
 from pinecone.core.utils.constants import MAX_MSG_SIZE, REQUEST_ID, CLIENT_VERSION
+from pinecone.core.grpc.protos.vector_service_pb2 import Vector as GRPCVector, QueryVector as GRPCQueryVector, \
+    UpsertRequest, DeleteRequest, \
+    QueryRequest, FetchRequest, DescribeIndexStatsRequest
+from pinecone.core.utils.error_handling import validate_and_convert_errors
+from pinecone import Vector as _Vector
```

Comment:
Can you test doing an `import python` with this? This might be a circular import, where this module tries to import a symbol from the pinecone module while that module is still half-initialized and trying to import from here. If so, import pinecone will fail.

You should be able to just use the pinecone.core.client.models package instead.